File: frontend/apps/ios/Pantopus/Features/ContentDetail/GigQuestionsSection.swift
//
//  GigQuestionsSection.swift
//  Pantopus
//
//  Structured Q&A block on Task Details — ask form, thread list, and
//  poster answer affordance. Matches RN/web `QASection`.
//

import SwiftUI

struct GigQuestionsSection: View {
    @Bindable var viewModel: GigDetailViewModel
    var onError: (@MainActor (String) -> Void)?

    private let maxQuestionLength = 1000
    private let maxAnswerLength = 2000

    var body: some View {
        VStack(alignment: .leading, spacing: Spacing.s3) {
            header
            if viewModel.canAskQuestion {
                askForm
            }
            questionsBody
        }
        .padding(.horizontal, Spacing.s5)
        .padding(.top, 22)
        .accessibilityIdentifier("gigQuestionsSection")
    }

    private var header: some View {
        HStack {
            Text("Questions (\(viewModel.questions.count))")
                .font(.system(size: 15, weight: .bold))
                .foregroundStyle(Theme.Color.appText)
            Spacer()
            statusChips
        }
        .accessibilityIdentifier("gigQuestionsHeader")
    }

    @ViewBuilder private var statusChips: some View {
        let answered = viewModel.questions.filter(\.isAnswered).count
        let open = viewModel.questions.count - answered
        HStack(spacing: 6) {
            if answered > 0 {
                Text("\(answered) answered")
                    .font(.system(size: 10, weight: .semibold))
                    .foregroundStyle(Theme.Color.success)
                    .padding(.horizontal, 8)
                    .padding(.vertical, 3)
                    .background(Theme.Color.success.opacity(0.12))
                    .clipShape(Capsule())
            }
            if open > 0 {
                Text("\(open) awaiting")
                    .font(.system(size: 10, weight: .semibold))
                    .foregroundStyle(Theme.Color.warning)
                    .padding(.horizontal, 8)
                    .padding(.vertical, 3)
                    .background(Theme.Color.warning.opacity(0.12))
                    .clipShape(Capsule())
            }
        }
    }

    private var askForm: some View {
        VStack(alignment: .leading, spacing: Spacing.s2) {
            TextField(
                "Ask a question about this gig...",
                text: $viewModel.newQuestionText,
                axis: .vertical
            )
            .lineLimit(2...4)
            .font(.system(size: 13.5))
            .foregroundStyle(Theme.Color.appText)
            .accessibilityIdentifier("gigQuestionsAskInput")
            HStack {
                Text("\(viewModel.newQuestionText.count)/\(maxQuestionLength)")
                    .font(.system(size: 11))
                    .foregroundStyle(Theme.Color.appTextMuted)
                Spacer()
                Button {
                    Task {
                        if let message = await viewModel.submitQuestion() {
                            onError?(message)
                        }
                    }
                } label: {
                    Text(viewModel.questionSubmitting ? "Posting…" : "Ask Question")
                        .font(.system(size: 13, weight: .semibold))
                        .foregroundStyle(Theme.Color.appTextInverse)
                        .padding(.horizontal, 14)
                        .padding(.vertical, 8)
                        .background(canSubmitQuestion ? Theme.Color.primary600 : Theme.Color.appSurfaceSunken)
                        .clipShape(RoundedRectangle(cornerRadius: Radii.md, style: .continuous))
                }
                .disabled(!canSubmitQuestion || viewModel.questionSubmitting)
                .buttonStyle(.plain)
                .accessibilityIdentifier("gigQuestionsAskButton")
            }
        }
        .padding(Spacing.s3)
        .background(Theme.Color.appSurface)
        .overlay(
            RoundedRectangle(cornerRadius: Radii.lg, style: .continuous)
                .stroke(Theme.Color.appBorder, lineWidth: 1)
        )
        .clipShape(RoundedRectangle(cornerRadius: Radii.lg, style: .continuous))
        .onChange(of: viewModel.newQuestionText) { _, newValue in
            if newValue.count > maxQuestionLength {
                viewModel.newQuestionText = String(newValue.prefix(maxQuestionLength))
            }
        }
    }

    private var canSubmitQuestion: Bool {
        viewModel.newQuestionText.trimmingCharacters(in: .whitespacesAndNewlines).count >= 5
    }

    @ViewBuilder private var questionsBody: some View {
        if viewModel.questionsLoading {
            Text("Loading questions...")
                .font(.system(size: 13))
                .foregroundStyle(Theme.Color.appTextSecondary)
                .frame(maxWidth: .infinity)
                .padding(.vertical, Spacing.s4)
        } else if viewModel.questions.isEmpty {
            Text("No questions yet. Be the first to ask!")
                .font(.system(size: 13))
                .foregroundStyle(Theme.Color.appTextSecondary)
                .multilineTextAlignment(.center)
                .frame(maxWidth: .infinity)
                .padding(.vertical, Spacing.s4)
                .accessibilityIdentifier("gigQuestionsEmpty")
        } else {
            VStack(spacing: Spacing.s3) {
                ForEach(viewModel.questions) { question in
                    questionRow(question)
                }
            }
        }
    }

    private func questionRow(_ question: GigQuestionDTO) -> some View {
        VStack(alignment: .leading, spacing: Spacing.s2) {
            Text(question.question)
                .font(.system(size: 13.5, weight: .semibold))
                .foregroundStyle(Theme.Color.appText)
            metaRow(question)
            if let answer = question.answer, !answer.isEmpty {
                answerBlock(question, answer: answer)
            }
            if viewModel.viewerIsOwner, !question.isAnswered {
                ownerAnswerControls(question)
            }
        }
        .padding(Spacing.s3)
        .background(Theme.Color.appSurface)
        .overlay(
            RoundedRectangle(cornerRadius: Radii.lg, style: .continuous)
                .stroke(Theme.Color.appBorder, lineWidth: 1)
        )
        .clipShape(RoundedRectangle(cornerRadius: Radii.lg, style: .continuous))
        .accessibilityIdentifier("gigQuestion.\(question.id)")
    }

    private func metaRow(_ question: GigQuestionDTO) -> some View {
        HStack(spacing: 6) {
            Text(askerDisplayName(question.asker))
                .font(.system(size: 11.5, weight: .medium))
                .foregroundStyle(Theme.Color.appTextSecondary)
            if let when = questionRelativeTime(question.createdAt) {
                Text("·")
                    .foregroundStyle(Theme.Color.appTextMuted)
                Text(when)
                    .font(.system(size: 11.5))
                    .foregroundStyle(Theme.Color.appTextMuted)
            }
            if question.isAnswered {
                Text("·")
                    .foregroundStyle(Theme.Color.appTextMuted)
                Text("Answered")
                    .font(.system(size: 11.5, weight: .semibold))
                    .foregroundStyle(Theme.Color.success)
            } else {
                Text("·")
                    .foregroundStyle(Theme.Color.appTextMuted)
                Text("Awaiting answer")
                    .font(.system(size: 11.5, weight: .medium))
                    .foregroundStyle(Theme.Color.warning)
            }
        }
    }

    private func answerBlock(_ question: GigQuestionDTO, answer: String) -> some View {
        VStack(alignment: .leading, spacing: 4) {
            Text("\(answererDisplayName(question)) answered:")
                .font(.system(size: 11.5, weight: .semibold))
                .foregroundStyle(Theme.Color.success)
            Text(answer)
                .font(.system(size: 13.5))
                .foregroundStyle(Theme.Color.appTextStrong)
        }
        .padding(Spacing.s2)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(Theme.Color.success.opacity(0.08))
        .overlay(alignment: .leading) {
            Rectangle()
                .fill(Theme.Color.success)
                .frame(width: 2)
        }
        .clipShape(RoundedRectangle(cornerRadius: Radii.sm, style: .continuous))
    }

    @ViewBuilder private func ownerAnswerControls(_ question: GigQuestionDTO) -> some View {
        if viewModel.answeringQuestionId == question.id {
            VStack(alignment: .leading, spacing: Spacing.s2) {
                TextField("Write your answer...", text: $viewModel.answerDraftText, axis: .vertical)
                    .lineLimit(2...4)
                    .font(.system(size: 13.5))
                    .padding(Spacing.s2)
                    .background(Theme.Color.appSurfaceSunken)
                    .clipShape(RoundedRectangle(cornerRadius: Radii.md, style: .continuous))
                    .accessibilityIdentifier("gigQuestionsAnswerInput")
                    .onChange(of: viewModel.answerDraftText) { _, newValue in
                        if newValue.count > maxAnswerLength {
                            viewModel.answerDraftText = String(newValue.prefix(maxAnswerLength))
                        }
                    }
                HStack {
                    Button("Cancel") { viewModel.cancelAnswering() }
                        .font(.system(size: 12, weight: .semibold))
                        .foregroundStyle(Theme.Color.appTextSecondary)
                        .buttonStyle(.plain)
                    Spacer()
                    Button {
                        Task {
                            if let message = await viewModel.submitAnswer(questionId: question.id) {
                                onError?(message)
                            }
                        }
                    } label: {
                        Text(viewModel.answerSubmitting ? "Posting…" : "Post Answer")
                            .font(.system(size: 12, weight: .semibold))
                            .foregroundStyle(Theme.Color.appTextInverse)
                            .padding(.horizontal, 12)
                            .padding(.vertical, 6)
                            .background(
                                viewModel.answerDraftText.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
                                    ? Theme.Color.appSurfaceSunken
                                    : Theme.Color.success
                            )
                            .clipShape(RoundedRectangle(cornerRadius: Radii.md, style: .continuous))
                    }
                    .disabled(
                        viewModel.answerDraftText.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
                            || viewModel.answerSubmitting
                    )
                    .buttonStyle(.plain)
                    .accessibilityIdentifier("gigQuestionsAnswerSubmit")
                }
            }
        } else {
            Button("Answer") { viewModel.beginAnswering(question.id) }
                .font(.system(size: 12, weight: .semibold))
                .foregroundStyle(Theme.Color.primary600)
                .buttonStyle(.plain)
                .accessibilityIdentifier("gigQuestionsAnswerButton")
        }
    }
}

private func askerDisplayName(_ user: GigQuestionUser?) -> String {
    guard let user else { return "Neighbor" }
    if let name = user.name, !name.isEmpty { return name }
    let parts = [user.firstName, user.lastName].compactMap { $0 }.filter { !$0.isEmpty }
    if !parts.isEmpty { return parts.joined(separator: " ") }
    return user.username ?? "Neighbor"
}

private func answererDisplayName(_ question: GigQuestionDTO) -> String {
    if let display = question.answererDisplayName, !display.isEmpty { return display }
    if let answerer = question.answerer {
        if let name = answerer.name, !name.isEmpty { return name }
        if let username = answerer.username, !username.isEmpty { return username }
    }
    return "Poster"
}

private func questionRelativeTime(_ iso: String?) -> String? {
    guard let iso else { return nil }
    let parser = ISO8601DateFormatter()
    parser.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
    let date = parser.date(from: iso) ?? ISO8601DateFormatter().date(from: iso)
    guard let date else { return nil }
    let interval = Date().timeIntervalSince(date)
    if interval < 60 { return "now" }
    if interval < 3600 { return "\(Int(interval / 60))m ago" }
    if interval < 86400 { return "\(Int(interval / 3600))h ago" }
    if interval < 604_800 { return "\(Int(interval / 86400))d ago" }
    return "\(Int(interval / 604_800))w ago"
}

File: frontend/apps/android/app/src/main/java/app/pantopus/android/ui/screens/contentdetail/GigQuestionsSection.kt
@file:Suppress("MagicNumber", "PackageNaming", "LongMethod")

package app.pantopus.android.ui.screens.contentdetail

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.BasicTextField
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.SolidColor
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import app.pantopus.android.data.api.models.gigs.GigQuestionDto
import app.pantopus.android.data.api.models.gigs.GigQuestionUser
import app.pantopus.android.ui.theme.PantopusColors
import app.pantopus.android.ui.theme.Radii
import app.pantopus.android.ui.theme.Spacing
import java.time.Duration
import java.time.Instant

@Composable
fun GigQuestionsSection(
    viewModel: GigDetailViewModel,
    onError: (String) -> Unit = {},
) {
    val questions by viewModel.questions.collectAsStateWithLifecycle()
    val loading by viewModel.questionsLoading.collectAsStateWithLifecycle()
    val newQuestion by viewModel.newQuestionText.collectAsStateWithLifecycle()
    val answeringId by viewModel.answeringQuestionId.collectAsStateWithLifecycle()
    val answerDraft by viewModel.answerDraftText.collectAsStateWithLifecycle()
    val questionSubmitting by viewModel.questionSubmitting.collectAsStateWithLifecycle()
    val answerSubmitting by viewModel.answerSubmitting.collectAsStateWithLifecycle()
    val canAsk = viewModel.canAskQuestion()
    val viewerIsOwner = viewModel.viewerIsOwner()

    Column(
        modifier =
            Modifier
                .fillMaxWidth()
                .padding(horizontal = Spacing.s5)
                .padding(top = 22.dp)
                .testTag("gigQuestionsSection"),
        verticalArrangement = Arrangement.spacedBy(Spacing.s3),
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(
                text = "Questions (${questions.size})",
                fontSize = 15.sp,
                fontWeight = FontWeight.Bold,
                color = PantopusColors.appText,
                modifier = Modifier.testTag("gigQuestionsHeader"),
            )
            QuestionStatusChips(questions = questions)
        }
        if (canAsk) {
            AskQuestionForm(
                text = newQuestion,
                submitting = questionSubmitting,
                onTextChange = viewModel::setNewQuestionText,
                onSubmit = { viewModel.submitQuestion(onError) },
            )
        }
        when {
            loading ->
                Text(
                    text = "Loading questions...",
                    fontSize = 13.sp,
                    color = PantopusColors.appTextSecondary,
                    modifier = Modifier.fillMaxWidth().padding(vertical = Spacing.s4),
                )
            questions.isEmpty() ->
                Text(
                    text = "No questions yet. Be the first to ask!",
                    fontSize = 13.sp,
                    color = PantopusColors.appTextSecondary,
                    modifier =
                        Modifier
                            .fillMaxWidth()
                            .padding(vertical = Spacing.s4)
                            .testTag("gigQuestionsEmpty"),
                )
            else ->
                Column(verticalArrangement = Arrangement.spacedBy(Spacing.s3)) {
                    questions.forEach { question ->
                        QuestionRow(
                            question = question,
                            viewerIsOwner = viewerIsOwner,
                            answeringId = answeringId,
                            answerDraft = answerDraft,
                            answerSubmitting = answerSubmitting,
                            onBeginAnswer = viewModel::beginAnswering,
                            onCancelAnswer = viewModel::cancelAnswering,
                            onAnswerDraftChange = viewModel::setAnswerDraftText,
                            onSubmitAnswer = { viewModel.submitAnswer(question.id, onError) },
                        )
                    }
                }
        }
    }
}

@Composable
private fun QuestionStatusChips(questions: List<GigQuestionDto>) {
    val answered = questions.count { it.isAnswered }
    val open = questions.size - answered
    Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
        if (answered > 0) {
            StatusChip(text = "$answered answered", color = PantopusColors.success)
        }
        if (open > 0) {
            StatusChip(text = "$open awaiting", color = PantopusColors.warning)
        }
    }
}

@Composable
private fun StatusChip(
    text: String,
    color: androidx.compose.ui.graphics.Color,
) {
    Text(
        text = text,
        fontSize = 10.sp,
        fontWeight = FontWeight.SemiBold,
        color = color,
        modifier =
            Modifier
                .clip(RoundedCornerShape(Radii.pill))
                .background(color.copy(alpha = 0.12f))
                .padding(horizontal = 8.dp, vertical = 3.dp),
    )
}

@Composable
private fun AskQuestionForm(
    text: String,
    submitting: Boolean,
    onTextChange: (String) -> Unit,
    onSubmit: () -> Unit,
) {
    val canSubmit = text.trim().length >= 5 && !submitting
    Column(
        modifier =
            Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(Radii.lg))
                .background(PantopusColors.appSurface)
                .border(1.dp, PantopusColors.appBorder, RoundedCornerShape(Radii.lg))
                .padding(Spacing.s3),
        verticalArrangement = Arrangement.spacedBy(Spacing.s2),
    ) {
        BasicTextField(
            value = text,
            onValueChange = onTextChange,
            modifier = Modifier.fillMaxWidth().testTag("gigQuestionsAskInput"),
            textStyle =
                androidx.compose.ui.text.TextStyle(
                    color = PantopusColors.appText,
                    fontSize = 13.5.sp,
                ),
            cursorBrush = SolidColor(PantopusColors.primary600),
            decorationBox = { inner ->
                if (text.isEmpty()) {
                    Text(
                        text = "Ask a question about this gig...",
                        fontSize = 13.5.sp,
                        color = PantopusColors.appTextMuted,
                    )
                }
                inner()
            },
        )
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(
                text = "${text.length}/1000",
                fontSize = 11.sp,
                color = PantopusColors.appTextMuted,
            )
            Box(
                modifier =
                    Modifier
                        .clip(RoundedCornerShape(Radii.md))
                        .background(if (canSubmit) PantopusColors.primary600 else PantopusColors.appSurfaceSunken)
                        .clickable(enabled = canSubmit, onClick = onSubmit)
                        .padding(horizontal = 14.dp, vertical = 8.dp)
                        .testTag("gigQuestionsAskButton"),
            ) {
                Text(
                    text = if (submitting) "Posting…" else "Ask Question",
                    fontSize = 13.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = if (canSubmit) PantopusColors.appTextInverse else PantopusColors.appTextMuted,
                )
            }
        }
    }
}

@Composable
private fun QuestionRow(
    question: GigQuestionDto,
    viewerIsOwner: Boolean,
    answeringId: String?,
    answerDraft: String,
    answerSubmitting: Boolean,
    onBeginAnswer: (String) -> Unit,
    onCancelAnswer: () -> Unit,
    onAnswerDraftChange: (String) -> Unit,
    onSubmitAnswer: () -> Unit,
) {
    Column(
        modifier =
            Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(Radii.lg))
                .background(PantopusColors.appSurface)
                .border(1.dp, PantopusColors.appBorder, RoundedCornerShape(Radii.lg))
                .padding(Spacing.s3)
                .testTag("gigQuestion.${question.id}"),
        verticalArrangement = Arrangement.spacedBy(Spacing.s2),
    ) {
        Text(
            text = question.question,
            fontSize = 13.5.sp,
            fontWeight = FontWeight.SemiBold,
            color = PantopusColors.appText,
        )
        QuestionMeta(question)
        question.answer?.takeIf { it.isNotEmpty() }?.let { answer ->
            AnswerBlock(question = question, answer = answer)
        }
        if (viewerIsOwner && !question.isAnswered) {
            OwnerAnswerControls(
                questionId = question.id,
                answeringId = answeringId,
                answerDraft = answerDraft,
                answerSubmitting = answerSubmitting,
                onBeginAnswer = onBeginAnswer,
                onCancelAnswer = onCancelAnswer,
                onAnswerDraftChange = onAnswerDraftChange,
                onSubmitAnswer = onSubmitAnswer,
            )
        }
    }
}

@Composable
private fun QuestionMeta(question: GigQuestionDto) {
    Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
        Text(
            text = askerDisplayName(question.asker),
            fontSize = 11.5.sp,
            fontWeight = FontWeight.Medium,
            color = PantopusColors.appTextSecondary,
        )
        relativeTime(question.createdAt)?.let {
            Text(text = "·", color = PantopusColors.appTextMuted, fontSize = 11.5.sp)
            Text(text = it, fontSize = 11.5.sp, color = PantopusColors.appTextMuted)
        }
        Text(text = "·", color = PantopusColors.appTextMuted, fontSize = 11.5.sp)
        Text(
            text = if (question.isAnswered) "Answered" else "Awaiting answer",
            fontSize = 11.5.sp,
            fontWeight = if (question.isAnswered) FontWeight.SemiBold else FontWeight.Medium,
            color = if (question.isAnswered) PantopusColors.success else PantopusColors.warning,
        )
    }
}

@Composable
private fun AnswerBlock(
    question: GigQuestionDto,
    answer: String,
) {
    Column(
        modifier =
            Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(Radii.sm))
                .background(PantopusColors.success.copy(alpha = 0.08f))
                .padding(Spacing.s2),
        verticalArrangement = Arrangement.spacedBy(4.dp),
    ) {
        Text(
            text = "${answererDisplayName(question)} answered:",
            fontSize = 11.5.sp,
            fontWeight = FontWeight.SemiBold,
            color = PantopusColors.success,
        )
        Text(text = answer, fontSize = 13.5.sp, color = PantopusColors.appTextStrong)
    }
}

@Composable
private fun OwnerAnswerControls(
    questionId: String,
    answeringId: String?,
    answerDraft: String,
    answerSubmitting: Boolean,
    onBeginAnswer: (String) -> Unit,
    onCancelAnswer: () -> Unit,
    onAnswerDraftChange: (String) -> Unit,
    onSubmitAnswer: () -> Unit,
) {
    if (answeringId == questionId) {
        Column(verticalArrangement = Arrangement.spacedBy(Spacing.s2)) {
            BasicTextField(
                value = answerDraft,
                onValueChange = onAnswerDraftChange,
                modifier =
                    Modifier
                        .fillMaxWidth()
                        .clip(RoundedCornerShape(Radii.md))
                        .background(PantopusColors.appSurfaceSunken)
                        .padding(Spacing.s2)
                        .testTag("gigQuestionsAnswerInput"),
                textStyle =
                    androidx.compose.ui.text.TextStyle(
                        color = PantopusColors.appText,
                        fontSize = 13.5.sp,
                    ),
                cursorBrush = SolidColor(PantopusColors.primary600),
                decorationBox = { inner ->
                    if (answerDraft.isEmpty()) {
                        Text(
                            text = "Write your answer...",
                            fontSize = 13.5.sp,
                            color = PantopusColors.appTextMuted,
                        )
                    }
                    inner()
                },
            )
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Text(
                    text = "Cancel",
                    fontSize = 12.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = PantopusColors.appTextSecondary,
                    modifier = Modifier.clickable(onClick = onCancelAnswer),
                )
                val canSubmit = answerDraft.trim().isNotEmpty() && !answerSubmitting
                Box(
                    modifier =
                        Modifier
                            .clip(RoundedCornerShape(Radii.md))
                            .background(if (canSubmit) PantopusColors.success else PantopusColors.appSurfaceSunken)
                            .clickable(enabled = canSubmit, onClick = onSubmitAnswer)
                            .padding(horizontal = 12.dp, vertical = 6.dp)
                            .testTag("gigQuestionsAnswerSubmit"),
                ) {
                    Text(
                        text = if (answerSubmitting) "Posting…" else "Post Answer",
                        fontSize = 12.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = if (canSubmit) PantopusColors.appTextInverse else PantopusColors.appTextMuted,
                    )
                }
            }
        }
    } else {
        Text(
            text = "Answer",
            fontSize = 12.sp,
            fontWeight = FontWeight.SemiBold,
            color = PantopusColors.primary600,
            modifier =
                Modifier
                    .clickable { onBeginAnswer(questionId) }
                    .testTag("gigQuestionsAnswerButton"),
        )
    }
}

private fun askerDisplayName(user: GigQuestionUser?): String {
    if (user == null) return "Neighbor"
    user.name?.takeIf { it.isNotEmpty() }?.let { return it }
    val parts = listOfNotNull(user.firstName, user.lastName).filter { it.isNotEmpty() }
    if (parts.isNotEmpty()) return parts.joinToString(" ")
    return user.username ?: "Neighbor"
}

private fun answererDisplayName(question: GigQuestionDto): String {
    question.answererDisplayName?.takeIf { it.isNotEmpty() }?.let { return it }
    question.answerer?.name?.takeIf { it.isNotEmpty() }?.let { return it }
    question.answerer?.username?.takeIf { it.isNotEmpty() }?.let { return it }
    return "Poster"
}

private fun relativeTime(iso: String?): String? {
    if (iso.isNullOrEmpty()) return null
    return runCatching {
        val seconds = Duration.between(Instant.parse(iso), Instant.now()).seconds
        when {
            seconds < 60 -> "now"
            seconds < 3_600 -> "${seconds / 60}m ago"
            seconds < 86_400 -> "${seconds / 3_600}h ago"
            seconds < 604_800 -> "${seconds / 86_400}d ago"
            else -> "${seconds / 604_800}w ago"
        }
    }.getOrNull()
}
